### PR TITLE
Exit correctly when the launched process terminates quickly

### PR DESCRIPTION
**Pull request by giucam**
Task-Id: KDEND-124

### CODE REVIEW

File: tests/earlyexittest.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        // setting the probe is not strictly needed but we silence a runtime warning this way
+        options.setProbeABI(ProbeFinder::listProbeABIs().first());
+        options.setLaunchArguments(QStringList() << "sleep" << "1");
+        Launcher launcher(options);
+
+        QSignalSpy spy(&launcher, SIGNAL(finished()));
+
+        Q_ASSERT(launcher.start());
+
+        int loops = 0;
```
Comment (krf) [inline tests/earlyexittest.cpp]: Isn't that what `QSignalSpy::wait` does?

File: tests/earlyexittest.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/*
+  earlyexittest.cpp
+
+  This file is part of GammaRay, the Qt application inspection and
+  manipulation tool.
+
+  Copyright (C) 2016 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Volker Krause <volker.krause@kdab.com>
```
Comment (krf) [inline tests/earlyexittest.cpp]: You're the author ;)

File: launcher/launcher.cpp [hunk 3]
```diff
@@ -278,7 +278,7 @@ void Launcher::restartTimer()
 
 void Launcher::checkDone()
 {
-    if (d->state == Complete
+    if (d->state == InjectorFinished
```
Comment (krf) [inline launcher/launcher.cpp]: This expression now does some redundant checks: Right now it's `a || (b && a)` => Could be simplified to just `a`. Not sure if that's intended.

I'm not sure if this change is fine, that's probably something for @vkrause.

File: tests/earlyexittest.cpp (hunk 2 above)
Comment (giucam) [inline tests/earlyexittest.cpp]: Right :)

File: tests/earlyexittest.cpp (hunk 1 above)
Comment (giucam) [inline tests/earlyexittest.cpp]: Yes but it was added in 5.0, so for 4.8 we'd need this trick anyway.

File: launcher/launcher.cpp (hunk 3 above)
Comment (giucam) [inline launcher/launcher.cpp]: Right... no, it wasn't intended.

File: tests/earlyexittest.cpp (hunk 1 above)
Comment (krf) [inline tests/earlyexittest.cpp]: Ah, got it. I'd still use `QSignalSpy::wait` conditionally there so we can easily detect the custom code as soon as Qt5 will be a required dependency.

**Comment by giucam** [comment]
@krf Does the approval mean you are ok with this even without volker's ok or did you get it through other means?

**Pull request merged**